scripts/comment_generator.py

```diff
@@ -66,20 +67,25 @@ def get_url(archive_id, path_to_urls):
     comment = "Thank you for this contribution."
 
     if len(list(acceptance_test_report[NEW_ERRORS])) != 0:
-        comment = comment + " Due to changes in this pull request, the " \
-                            "following validation rules trigger new errors:\n"
+        comment = (
+            comment + " Due to changes in this pull request, the "
+            "following validation rules trigger new errors:\n"
+        )
 
         for notice_sample in acceptance_test_report[NEW_ERRORS]:
             notice_code = list(notice_sample.keys())[0]
             notice_info = f"- `{notice_code}`: {notice_sample[notice_code][AFFECTED_SOURCES_COUNT]} datasets (including "
             for source_ids in notice_sample[notice_code][AFFECTED_SOURCES]:
-                for source_id in list(source_ids.keys()):
-                    notice_info += (
-                        f"[`{source_id}`]({get_url(source_id, args.path_to_urls)}), "
-                    )
+                source_id = source_ids.get(SOURCE_ID_KEY)
+                notice_info += (
+                    f"[`{source_id}`]({get_url(source_id, args.path_to_urls)}), "
```

Comment:
you can use `urls_map` here instead of `get_url` that reads and parses the file every time